### PR TITLE
Adds quick grant view content

### LINKED ISSUE #1790: Item perm panel: add a quick add view perm button
## Overview
Add for manager an quick way to give view access to a group.

## Acceptance criteria
In the permission panel displayed on top of items when observing: 
If: 
  - [ ] the current user can give perm to the group
  - [ ] and the current user can grant item view >= content 
  - [ ] and the group has can view = none

Then:
- [ ] next to the "Can view none" text, add a button "Grant content access"
- [ ] When pressed, change the permission directly. 
- [ ] While processing, disable the button
- [ ] When completed: display a toast on success/failure, and refresh the item on success

### PR DESCRIPTION
**Pull request by Iloveall**
## Description

Fixes #1790 

## Notes (out of scope, known isues, hints for reviewing code, ...)  (optional)

During development I've faced randomly 400 error when created the item, so It could be potentially reason of failed tests 

`{
  "success": false,
  "message": "Bad Request",
  "error_text": "EOF"
}`

## Test cases

[Branch for test](https://dev.algorea.org/branch/feature/quick-add-perm-button/en/a/home;pa=0)

### CODE REVIEW

**Comment by smadbe** [comment]
[auto-generated message] E2E test report on: https://algorea-static.s3.eu-central-1.amazonaws.com/branch/feature/quick-add-perm-button/playwright-report/index.html

**Comment by smadbe** [comment]
[auto-generated message] E2E test report on: https://algorea-static.s3.eu-central-1.amazonaws.com/branch/feature/quick-add-perm-button/playwright-report/index.html

**Comment by smadbe** [comment]
I feel it is not very clear it is not a button and not just a badge... 
Do you have idea how we could improve that? Maybe already by adding a "+" icon on the left? 
Also, probably the button should be in orange probably, don't you think?

**Comment by smadbe** [comment]
Also the "can view" on hover is weird

**Comment by Iloveall** [comment]
> As discussed on Slack:
> 
> * remove the tooltip/title on the row which displays "can view" on hover
> * make the border color a bit darker on hover

- I've fix case with border color;
- For tooltip case I've created issue as it requires to do changes in other files that not related to current issue https://github.com/France-ioi/AlgoreaFrontend/issues/1820

File: src/app/items/containers/item-permissions/item-permissions.component.ts [hunk 1]
```diff
@@ -90,4 +100,33 @@ export class ItemPermissionsComponent implements OnChanges {
       this.changed.emit();
     }
   }
+
+  grantAccess(): void {
```
Comment (smadbe) [inline src/app/items/containers/item-permissions/item-permissions.component.ts:104]: "grantAccess" may be confusing with the other permission button. Call it "grantViewContentAccess" to be more explicit.

**Review by smadbe — APPROVED** [review]

File: src/app/items/containers/item-permissions/item-permissions.component.ts (hunk 1 above)
Comment (Iloveall) [reply src/app/items/containers/item-permissions/item-permissions.component.ts]: Fixed

**Comment by smadbe** [comment]
[auto-generated message] E2E test report on: https://algorea-static.s3.eu-central-1.amazonaws.com/branch/feature/quick-add-perm-button/playwright-report/index.html

**Pull request merged**